packages/api-derive/src/democracy/locks.ts

```diff
@@ -46,18 +46,25 @@ function delegateLocks (api: ApiInterfaceRx, { balance, conviction, target }: Vo
         referendumId,
         unlockAt: unlockAt.isZero()
           ? unlockAt
-          : referendumEnd.add(api.consts.democracy.enactmentPeriod.muln(LOCKUPS[conviction.index])),
-        vote: api.registry.createType('Vote', { aye: vote.isAye, conviction })
+          : referendumEnd?.add(api.consts.democracy.enactmentPeriod.muln(LOCKUPS[conviction.index])) || new BN(0),
+        vote: api.registry.createType('Vote', { aye: vote?.isAye, conviction })
       }))
     )
   );
 }
 
-function directLocks (api: ApiInterfaceRx, { votes }: VotingDirect): Observable<DeriveDemocracyLock[]> {
-  if (!votes.length) {
+function directLocks (api: ApiInterfaceRx, { prior, votes }: VotingDirect): Observable<DeriveDemocracyLock[]> {
+  if (!votes.length && !prior.length) {
     return of([]);
   }
 
+  if (prior.length) {
```

Comment:
I finally have a testcase for this - months later.

The above is not quite correct. So when unlocked the direct vote is there in all cases. So this path would always be followed.

- when a proxy is removed, it has non-zero values
- with just normal clearing, it has 0/0 values (both balance and unlockAt is 0)

So in the case with a proxy removed, it looks like -

![image](https://user-images.githubusercontent.com/1424473/115344011-ff069380-a1ac-11eb-80eb-9c1d7370adb9.png)

And in the case of stuff just cleared, it looks like

![image](https://user-images.githubusercontent.com/1424473/115344048-0e85dc80-a1ad-11eb-8514-d466e726e434.png)